### PR TITLE
Arman: Tasks_281-320

### PR DESCRIPTION
**Pull request by hsp-arman**

### CODE REVIEW

File: src/main/java/am/hsp/apps/seua/tasks_281_320/Task_300.java [hunk 1]
```diff
@@ -0,0 +1,91 @@
+package main.java.am.hsp.apps.seua.tasks_281_320;
+
+import main.java.am.hsp.apps.seua.base.TaskArray;
+
+/**
+ * 
+ * Print new array , without the elements, which
+ * square value is lower than comparable
```
Comment (hsp-harut) [inline src/main/java/am/hsp/apps/seua/tasks_281_320/Task_300.java]: less than
Remember: Greater >, Greater or Equal >=, Less <, Less or Equal <=

File: src/main/java/am/hsp/apps/seua/tasks_281_320/Task_300.java [hunk 2]
```diff
@@ -0,0 +1,91 @@
...
+    }
+
+    /**
+     * Return count of elements, without the elements, which square value is lower
+     * than comparable
+     * 
+     * @param array
+     * @return
+     */
+    private int getCountOfElements(int[] array, int comparable) {
```
Comment (hsp-harut) [inline src/main/java/am/hsp/apps/seua/tasks_281_320/Task_300.java]: same

File: src/main/java/am/hsp/apps/seua/tasks_281_320/Task_300.java [hunk 3]
```diff
@@ -0,0 +1,91 @@
...
+     * 
+     * @param array
+     * @return
+     */
+    private int[] getNewArray(int[] array, int comparable) {
+        int[] newArray = new int[getCountOfElements(array, comparable)];
+        int j = 0;
+        
+        for (int i = 0; i < array.length; i++) {
+            if (!isSquareOfNumberLowerThanComparable(array[i], comparable)) {
```
Comment (hsp-harut) [inline src/main/java/am/hsp/apps/seua/tasks_281_320/Task_300.java]: same

File: src/main/java/am/hsp/apps/seua/tasks_281_320/Task_312.java [hunk 4]
```diff
@@ -0,0 +1,57 @@
...
+     * 
+     * @param array
+     * @return
+     */
+    private int[] getNewArray(int[] array) {
+        int[] newArray = new int[array.length / 2];
+        int j = 0;
+
+        for (int i = 0; i < array.length; i += 2) {
+            newArray[j] = getMax(array[i], array[i + 1]);
```
Comment (hsp-harut) [inline src/main/java/am/hsp/apps/seua/tasks_281_320/Task_312.java]: Your program will throw an exception when your array contains Odd count of elements.

File: src/main/java/am/hsp/apps/seua/tasks_281_320/Task_314.java [hunk 5]
```diff
@@ -0,0 +1,89 @@
...
+     * 
+     * @param array
+     * @return
+     */
+    private int[] getNewArray(int[] array) {
+        int[] newArray = new int[getCountOfPositiveElements(array) + array.length];
+        int j = 0;
+
+        for (int i = 0; i < array.length; i++) {
+            newArray[j] = array[i];
```
Comment (hsp-harut) [inline src/main/java/am/hsp/apps/seua/tasks_281_320/Task_314.java]: newArray[j++]

File: src/main/java/am/hsp/apps/seua/tasks_281_320/Task_314.java [hunk 6]
```diff
@@ -0,0 +1,89 @@
...
+            if (isPositive(array[i])) {
+                newArray[j + 1] = 0;
```
Comment (hsp-harut) [inline src/main/java/am/hsp/apps/seua/tasks_281_320/Task_314.java]: newArray[j++]

File: src/main/java/am/hsp/apps/seua/tasks_281_320/Task_314.java [hunk 7]
```diff
@@ -0,0 +1,89 @@
...
+                j++;
+            }
+            j++;
```
Comment (hsp-harut) [inline src/main/java/am/hsp/apps/seua/tasks_281_320/Task_314.java]: remove this

File: src/main/java/am/hsp/apps/seua/tasks_281_320/Task_314.java [hunk 8]
```diff
@@ -0,0 +1,89 @@
...
+     */
+    private int[] getNewArray(int[] array) {
+        int[] newArray = new int[getCountOfPositiveElements(array) + array.length];
+        int j = 0;
+
+        for (int i = 0; i < array.length; i++) {
+            newArray[j] = array[i];
+            if (isPositive(array[i])) {
+                newArray[j + 1] = 0;
+                j++;
```
Comment (hsp-harut) [inline src/main/java/am/hsp/apps/seua/tasks_281_320/Task_314.java]: remove this

File: src/main/java/am/hsp/apps/seua/tasks_281_320/Task_281.java [hunk 9]
```diff
@@ -0,0 +1,85 @@
...
+
+    }
+
+    /**
+     * Create new array , using inly positive elements
+     * 
+     * @param array
+     * @return
+     */
+    private int[] getNewArray(int[] array) {
```
[the remark this answers is not in the document]
Comment (hsp-arman) [reply src/main/java/am/hsp/apps/seua/tasks_281_320/Task_281.java]: I think, getRequiredArray  is right

**Comment by hsp-arman** [comment]
Tasks : 
281,287,294,297, 300, 311, 312, 314, 317, 319

**Pull request merged**